File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.cc
#include <iostream>
using namespace std;

int main() {
  int n, count = 0;
  cin >> n;
  
  for (int k=1; k<n+1; k++) {
    int validation = 2*n - k*(k-1);
    if (validation <= 0) break;
    if (validation % (2*k) == 0) count++;
    }

    cout << count << '\n';
    
    return 0;
  }

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
N = int(input())
count = 0

for k in range(1, N+1):
  validation = (2*N)/k-k+1   #validation = 2a
  if validation <= 0:    #2N/k 에서 결론적으로 부동소수점 오차 발생해서 의미 없음
    break
  if validation % 2 == 0:
    count += 1

print(count)